checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,200 @@
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.Tree;
+
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.analysis.AnalysisResult;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeMirror;
+
+/** The type factory for the {@link InitializationFieldAccessSubchecker}. */
+public class InitializationFieldAccessAnnotatedTypeFactory
+        extends InitializationParentAnnotatedTypeFactory {
+
+    /**
+     * Create a new InitializationFieldAccessAnnotatedTypeFactory.
+     *
+     * @param checker the checker to which the new type factory belongs
+     */
+    public InitializationFieldAccessAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+    }
+
+    @Override
+    protected InitializationAnalysis createFlowAnalysis() {
+        return new InitializationAnalysis(checker, this);
+    }
+
+    @Override
+    protected void performFlowAnalysis(ClassTree classTree) {
+        // Only perform the analysis if initialization checking is turned on.
+        if (!checker.hasOption("assumeInitialized")) {
+            super.performFlowAnalysis(classTree);
+            ;
+        }
+    }
+
+    /**
+     * Returns the flow analysis.
+     *
+     * @return the flow analysis
+     * @see #getFlowResult()
+     */
+    InitializationAnalysis getAnalysis() {
+        return analysis;
+    }
+
+    /**
+     * Returns the result of the flow analysis. Invariant:
+     *
+     * <pre>
+     *  scannedClasses.get(c) == FINISHED for some class c &rArr; flowResult != null
+     * </pre>
+     *
+     * Note that flowResult contains analysis results for Trees from multiple classes which are
+     * produced by multiple calls to performFlowAnalysis.
+     *
+     * @return the result of the flow analysis
+     * @see #getAnalysis()
+     */
+    AnalysisResult<CFValue, InitializationStore> getFlowResult() {
+        return flowResult;
+    }
+
+    /**
+     * This annotator should be added to {@link GenericAnnotatedTypeFactory#createTreeAnnotator} for
+     * the target checker. It ensures that the fields of an uninitialized receiver have the top type
+     * in the parent checker's hierarchy.
+     *
+     * @see InitializationChecker#getTargetCheckerClass()
+     */
+    public static class CommitmentFieldAccessTreeAnnotator extends TreeAnnotator {
```

Comment:
Would it be nicer to make this a top-level class? It is referred to separately by the target checker, right? Should the class be named `InitializationFieldAccessTreeAnnotator`? Or what is the reason for the `Commitment` vs `Initialization` naming?